### PR TITLE
Clarify when to use the *InContainer() functions

### PR DESCRIPTION
**Pull request by Vinnl**
A developer was unclear about whether to use `saveSolidDatasetInContainer` rather than just `saveSolidDatasetAt`; this PR adds some clarification on why you'd use the former, ever.

- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/resource/file.ts [hunk 1]
```diff
@@ -139,6 +139,16 @@ type SaveFileOptions = WriteFileOptions & {
  * be rejected. You can initialise it first using [[createContainerAt]], or directly save the file
  * at the desired location using [[overwriteFile]].
  *
+ * This function is primarily useful if the current user does not access to change existing files in
```
Comment (pmcb55) [inline src/resource/file.ts:142]:
```suggestion
 * This function is primarily useful if the current user does not have access to change existing files in
```

File: src/resource/file.ts [hunk 2]
```diff
@@ -139,6 +139,16 @@ type SaveFileOptions = WriteFileOptions & {
...
+ * a Container, but is allowed to add new files; in other words, they have Append, but not Write
+ * access to a Container. This is useful in situations where someone wants to allow others to e.g.
```
Comment (pmcb55) [inline src/resource/file.ts:144]: Personally I don't like the use of 'e.g.' in the middle of a sentence (it's Ok at the end of a sentence, or in brackets), so I'd just tweak this to:
```suggestion
 * access to a Container. This is useful in situations where someone wants to allow others to, for example,
```

File: src/resource/file.ts [hunk 3]
```diff
@@ -139,6 +139,16 @@ type SaveFileOptions = WriteFileOptions & {
...
+ * send notifications to their Pod, but not to view or delete existing ones.
```
Comment (pmcb55) [inline src/resource/file.ts:145]:
```suggestion
 * send notifications to their Pod, but not to view or delete existing notifications.
```

**Review by pmcb55 — APPROVED** [review]
Looks good - just some suggested tweaks.

File: src/resource/file.ts (hunk 1 above)
Comment (Vinnl) [reply src/resource/file.ts]: Incorporated this and the other suggestions in all three instances of this doc.

**Pull request merged**